src/test/unit_tests/test_bucketspace.js

```diff
@@ -1193,3 +1192,138 @@ mocha.describe('nsfs account configurations', function() {
+            email: EMAIL,
+        });
+        return {
+            access_key: account.access_keys[0].access_key.unwrap(),
+            secret_key: account.access_keys[0].secret_key.unwrap()
+        };
+    }
+    const random_name = (Math.random() + 1).toString(36).substring(7);
+    const nsfs_account_config = {
+        uid: uid || process.getuid(),
+        gid: gid || process.getgid(),
+        new_buckets_path: new_buckets_path || '/',
+        nsfs_only: nsfs_only || false
+    };
+
+    let account = await rpc_client.account.create_account({
+        has_login: false,
+        s3_access: true,
+        email: `${random_name}@noobaa.com`,
+        name: random_name,
+        nsfs_account_config
+    });
+    return {
+        access_key: account.access_keys[0].access_key.unwrap(),
+        secret_key: account.access_keys[0].secret_key.unwrap()
+    };
+}
+
+mocha.describe('namespace_fs - versioning', function() {
+    const nsr = 'versioned-nsr';
+    const bucket_name = 'versioned-bucket';
+    const tmp_fs_root = '/tmp/test_bucket_namespace_fs_versioning';
+    const bucket_path = '/bucket';
+    let s3_uid5;
+    let s3_uid6;
+    let s3_admin;
+
+    mocha.before(function() {
+        if (process.getgid() !== 0 || process.getuid() !== 0) {
+            coretest.log('No Root permissions found in env. Skipping test');
+            this.skip(); // eslint-disable-line no-invalid-this
+        }
+    });
+    const full_path = tmp_fs_root + bucket_path;
+    mocha.before(async () => fs_utils.create_fresh_path(tmp_fs_root, 0o777));
+    mocha.before(async () => fs_utils.create_fresh_path(full_path, 0o770));
+    mocha.after(async () => fs_utils.folder_delete(tmp_fs_root));
+
+    mocha.it('export dir as bucket', async function() {
+        await rpc_client.pool.create_namespace_resource({
+            name: nsr,
+            nsfs_config: {
+                fs_root_path: tmp_fs_root,
+                fs_backend: 'GPFS'
+            }
+        });
+        const obj_nsr = { resource: nsr, path: bucket_path };
+        await rpc_client.bucket.create_bucket({
+            name: bucket_name,
+            namespace: {
+                read_resources: [obj_nsr],
+                write_resource: obj_nsr
+            }
+        });
+
+        const policy = {
+            Version: '2012-10-17',
+            Statement: [{
+                Sid: 'id-1',
+                Effect: 'Allow',
+                Principal: { AWS: "*" },
+                Action: ['s3:PutBucketVersioning'],
+                Resource: [`arn:aws:s3:::${bucket_name}`]
+            }]
+        };
+        const { access_key, secret_key } = await generate_nsfs_account({ admin: true });
+        s3_admin = generate_s3_client(access_key, secret_key);
+        await s3_admin.putBucketPolicy({
+            Bucket: bucket_name,
+            Policy: JSON.stringify(policy)
+        }).promise();
+    });
+
+    mocha.it('create account uid, gid 5 - no permissions', async function() {
```

Comment:
why is this a test case? looks like a test setup

Reply:
moved it to the before section